test/dialects/ansi_test.py

```diff
@@ -163,12 +164,12 @@ def test__dialect__ansi_specific_segment_not_match(
         ("SELECT 1 + (2 ", [(1, 12)]),
         # Set expression with inappropriate ORDER BY or LIMIT. Error
         # raised on the UNION.
-        ("SELECT * FROM a ORDER BY 1 UNION SELECT * FROM b", [(1, 27)]),
-        ("SELECT * FROM a LIMIT 1 UNION SELECT * FROM b", [(1, 24)]),
-        ("SELECT * FROM a ORDER BY 1 LIMIT 1 UNION SELECT * FROM b", [(1, 35)]),
+        ("SELECT * FROM a ORDER BY 1 UNION SELECT * FROM b", [(1, 28)]),
+        ("SELECT * FROM a LIMIT 1 UNION SELECT * FROM b", [(1, 25)]),
+        ("SELECT * FROM a ORDER BY 1 LIMIT 1 UNION SELECT * FROM b", [(1, 36)]),
```

Comment:
NOTE: The only change here is that the unparsable segment now starts at the code element, not the whitespace before. I think this is an improvement.